### PR TITLE
add documents and notebooks

### PR DESCRIPTION
**Pull request by sangyx**

### CODE REVIEW

**Review by hameerabbasi — COMMENTED** [review]
Can we:
1. Test the notebooks?
2. Move all the images into `docs/_static`: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path
3. Add docs to CI?

**Comment by hameerabbasi** [comment]
Great work! It looks good to me, just a couple of small requests.

File: docs/auto_diff.rst [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+The optimization process of deep learning models is based on the gradient descent method. Deep learning frameworks such as PyTorch and Tensorflow can be divided into three parts: model api, gradient calculation and gpu acceleration. Gradient calculation plays an important role, and the core technology of this part is automatic differentiation.
+
+Differential Methods
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are four differential methods:
+
+* Manual differentiation
+* Numerical differentiation
+* Sigmbolic differential
```
Comment (dhirschfeld) [inline docs/auto_diff.rst]: ?
```suggestion
* Symbolic differentiation
```

Comment (hameerabbasi) [reply docs/auto_diff.rst]: Nice catch.

**Comment by sangyx** [comment]
> 1. Test the notebooks?

Could you give an example? I run the code and there is no bug.

> 3. Add docs to CI?

I have added `job: Docs` in azure-pipelines.yml.

**Comment by hameerabbasi** [comment]
You can add this PyTest plugin and add it to CI: https://github.com/computationalmodelling/nbval

**Pull request merged**